web3/pm.py

```diff
@@ -17,11 +24,98 @@
         pkg = Package(manifest, self.web3)
         return pkg
 
     def get_package_from_uri(self, uri):
         pkg = Package.from_uri(uri, self.web3)
         return pkg
+
+    def release_package(self, name, version, uri):
+        if not self.registry:
+            raise AttributeError("Registry has not been set for web3.pm")
+        if self.web3.eth.defaultAccount != self.registry.owner:
+            # change error
+            raise AttributeError("Owner doesn't have right permissions to release pkg.")
+        self.registry.release(name, version, uri)
+
+    def get_release_data_from_registry(self, name, version):
+        if not self.registry:
+            raise AttributeError("Registry has not been set for web3.pm")
+        return self.registry.get_release_data(name, version) 
+
+    def get_package_from_registry(self, name, version):
+        if not self.registry:
+            raise AttributeError("Registry has not been set for web3.pm")
+        release_uri = self.registry.get_release_data(name, version)
+        return self.get_package_from_uri(release_uri)
+
+
+class Registry(Contract):
+    def __init__(self, address, w3):
+        # only works with v.py registry in ethpm/assets
+        # abi verification for compatibility?
+        # todo ens linking
+        manifest = json.loads((ASSETS_DIR / 'vyper_registry' / '1.0.0.json').read_text())
+        registry_package = Package(manifest, w3)
+        self.registry = registry_package.get_contract_instance("registry", to_canonical_address(address))
+        self.w3 = w3
+        self.address = address
+
+    @classmethod
+    def deploy_new_instance(cls, w3):
+        manifest = json.loads((ASSETS_DIR / 'vyper_registry' / '1.0.0.json').read_text())
+        registry_package = Package(manifest, w3)
+        registry_deployer = Deployer(registry_package)
+        deployed_registry_package = registry_deployer.deploy("registry")
+        deployed_registry_address = deployed_registry_package.deployments.get_contract_instance("registry").address
+        return cls(deployed_registry_address, deployed_registry_package.w3)
+
+
+    def release(self, name, version, uri):
+        tx_receipt = self.registry.functions.release(name, version, uri).transact()
+        self.w3.eth.waitForTransactionReceipt(tx_receipt)
+        
+
+    def get_release_data(self, name, version):
+        return self.registry.functions.get_release_data(name, version).call()
+
+    @property
+    def owner(self):
+        return self.registry.functions.owner().call()
+
+    @to_tuple
+    def get_all_packages(self):
+        package_count = self.registry.functions.package_count().call()
+        for index in range(package_count):
+            package_id = self.registry.functions.get_package_id(index).call()
+            package_data = self.registry.functions.get_package_data_by_id(package_id).call()
+            yield (package_data[0].rstrip(b'\x00'), package_data[1])
```

Comment:
I think it might be nice for future maintainers if you decomposed `package_data` into it's two parts.  I am pretty sure `package_data[0]` is the package name.  It isn't clear to me what `package_data[1]` is.

Also, commenting on the reason for the `rstrip` call would be good.